docs/bundler/migration.md

```diff
@@ -0,0 +1,1107 @@
+    // when string, this is equivalent to entryNames
+    naming: "[name].[ext]",
+
+    // granular naming options
+    naming: {
+      entry: "[name].[ext]",
+      asset: "[name].[ext]",
+      chunk: "[name].[ext]",
+    },
+  });
+  ```
+
+---
+
+- `entryPoints`
+- `entrypoints`
+- Capitalization difference
+
+---
+
+- `external`
+- `external`
+- No differences
+
+---
+
+- `footer`
+- n/a
+- Not supported
+
+---
+
+- `format`
+- `format`
+- Only supports `"esm"` currently. Support for `"cjs"` and `"iife"` is planned.
+
+---
+
+- `globalName`
+- n/a
+- Not supported
+
+---
+
+- `ignoreAnnotations`
+- n/a
+- Not supported
+
+---
+
+- `inject`
+- n/a
+- Not supported
+
+---
+
+- `jsx`
+- `jsx`
+- Not supported in JS API, configure in `tsconfig.json`
+
+---
+
+- `jsxDev`
+- `jsxDev`
+- Not supported in JS API, configure in `tsconfig.json`
+
+---
+
+- `jsxFactory`
+- `jsxFactory`
+- Not supported in JS API, configure in `tsconfig.json`
+
+---
+
+- `jsxFragment`
+- `jsxFragment`
+- Not supported in JS API, configure in `tsconfig.json`
+
+---
+
+- `jsxImportSource`
+- `jsxImportSource`
+- Not supported in JS API, configure in `tsconfig.json`
+
+---
+
+- `jsxSideEffects`
+- `jsxSideEffects`
+- Not supported in JS API, configure in `tsconfig.json`
+
+---
+
+- `keepNames`
+- n/a
+- Not supported
+
+---
+
+- `legalComments`
+- n/a
+- Not supported
+
+---
+
+- `loader`
+- n/a
+- Not supported in JS API
+
+---
+
+- `logLevel`
+- n/a
+- Not supported
+
+---
+
+- `logLimit`
+- n/a
+- Not supported
+
+---
+
+- `logOverride`
+- n/a
+- Not supported
+
+---
+
+- `mainFields`
+- n/a
+- Not supported
+
+---
+
+- `mangleCache`
+- n/a
+- Not supported
+
+---
+
+- `mangleProps`
+- n/a
+- Not supported
+
+---
+
+- `mangleQuoted`
+- n/a
+- Not supported
+
+---
+
+- `metafile`
+- `manifest`
+- When `manifest` is `true`, the result of `Bun.build()` will contain a `manifest` property. The manifest is compatible with esbuild's metafile format.
+
+---
+
+- `minify`
+- `minify`
+- In Bun, `minify` can be a boolean or an object.
+
+  ```ts
+  Bun.build({
+    entrypoints: ['./index.tsx'],
+    // enable all minification
+    minify: true
+
+    // granular options
+    minify: {
+      identifiers: true,
+      syntax: true,
+      whitespace: true
+    }
+  })
+  ```
+
+---
+
+- `minifyIdentifiers`
+- `minify.identifiers`
+- Nested object syntax
+
+---
+
+- `minifySyntax`
+- `minify.syntax`
+- Nested object syntax
+
+---
+
+- `minifyWhitespace`
+- `minify.whitespace`
+- Nested object syntax
+
+---
+
+- `nodePaths`
+- n/a
+- Not supported
+
+---
+
+- `outExtension`
+- n/a
+- Not supported
+
+---
+
+- `outbase`
+- `root`
+- Different name
+
+---
+
+- `outdir`
+- `outdir`
+- No differences
+
+---
+
+- `outfile`
+- `outfile`
+- No differences
+
+---
+
+- `packages`
+- n/a
+- Not supported, use `external`
+
+---
+
+- `platform`
+- `target`
+- Supports `"bun"`, `"node"`, and `"browser"` (the default)
+
+---
+
+- `plugins`
+- `plugins`
+- Bun's plugin API is esbuild-compatible
```

Comment:
its not fully esbuild compatible. we could say something along the lines of "some of the api is implemented, whole thing will be sometime, see [link to the plugin compat table] for what is supported and not supported"